activerecord/test/cases/finder_test.rb

```diff
@@ -289,6 +289,30 @@ def test_exists_returns_false_with_false_arg
     assert_equal false, Topic.exists?(false)
   end
 
+  def test_exists_with_loaded_relation
+    topics = Topic.all.load
+    assert_no_queries do
+      assert_predicate topics, :exists?
+    end
+  end
+
+  def test_exists_with_empty_loaded_relation
+    Topic.delete_all
+    topics = Topic.all.load
+    assert_no_queries do
+      assert_not_predicate topics, :exists?
+    end
+  end
+
+  def test_exists_with_loaded_relation_having_unsaved_records
+    author = authors(:david)
+    posts = author.posts.load
+    assert_not_empty posts
+    posts.each(&:destroy)
+
+    assert_not_predicate posts, :exists?
```

Comment:
Sorry to ask. so should we add `assert_no_queries` as well here? @fatkodima

Reply:
yes, it would be good if it was added initially :)